### PR TITLE
Update candidates anonymization

### PR DESCRIPTION
**Pull request by abessette**

### CODE REVIEW

File: interview/views.py [hunk 1]
```diff
@@ -549,18 +548,12 @@ def new_candidate(request, past_candidate_id=None):
         interviewers_form = InterviewersForm(prefix="interviewers", data=request.POST)
 
         #  clicked on "reuse this candidate"
-        if not "summit" in request.POST:
+        if "reuse-candidate" in request.POST and past_candidate_id:
             candidate = Candidate.objects.get(id=past_candidate_id)
-            if not candidate.name:
```
Comment (bport) [inline interview/views.py:554]: We need to keep those checks for merging candidates

i.e. if the existing one contains phone number and not the new one, we want to keep this information from the old one

**Review by bport — CHANGES_REQUESTED** [review]

File: interview/views.py (hunk 1 above)
Comment (abessette) [reply interview/views.py:554]: Thank you for spotting this.
Please find below an associated fix.

File: interview/models.py [hunk 2]
```diff
@@ -381,6 +382,14 @@ def __str__(self):
             candidate=self.candidate, for_subsidiary=_("for subsidiary"), subsidiary=self.subsidiary
         )
 
+    def anonymize(self):
+        """
+        Anonymize the candidate process data, removing its extra information
+        This is irreversible
+        """
+        self.other_informations = ""
+        self.save()
```
Comment (bport) [inline interview/models.py:391]: Don't save both here and on caller code

File: interview/management/commands/anonymize.py [hunk 3]
```diff
@@ -31,17 +31,25 @@ def handle(self, *args, **options):
...
+            candidate.anonymize()
+            candidate.save()
+
+            for proc in candidate.process_set.all():
+                for interview in proc.interview_set.all():
+                    interview.anonymize()
+                    interview.save()
+
+                proc.anonymize()
+                proc.save()
```
Comment (bport) [inline interview/management/commands/anonymize.py:53]: We need to prevent email triggering

**Review by bport — CHANGES_REQUESTED** [review]

File: interview/models.py [hunk 4]
```diff
@@ -575,11 +575,12 @@ def save(self, *args, **kwargs):
...
         ):
             self.state = self.PLANNED
-            self.trigger_notification()
+            if trigger_notification:
+                self.trigger_notification()
         elif self.state == self.PLANNED and self.planned_date is None:
             self.state = self.WAITING_PLANIFICATION
 
-        super(Interview, self).save(*args, **kwargs)
+        super(Interview, self).save(force_insert=False, force_update=False, using=None, update_fields=None)
```
Comment (bport) [inline interview/models.py:583]: We don't want to force value to save method

**Pull request merged**